src/app-logic/url-handling.js

```diff
@@ -895,6 +954,52 @@ const _upgraders = {
       .filter(Boolean)
       .join('~');
   },
+  [6]: ({ query }: ProcessedLocationBeforeUpgrade) => {
+    // The tracks-related query arguments have been converted to use uintarray-encoding.
+    // Update them from the 0-10-9-8-1-2-3-4-5-6-7 syntax to the "0aw81w7" syntax.
+    if (query.globalTrackOrder) {
+      const globalTrackOrder = (query.globalTrackOrder: string)
+        .split('-')
+        .map(s => +s);
+      query.globalTrackOrder =
+        encodeUintArrayForUrlComponent(globalTrackOrder) || undefined;
+    }
+    if (query.hiddenGlobalTracks) {
+      const hiddenGlobalTracks = new Set(
+        (query.hiddenGlobalTracks: string).split('-').map(s => +s)
+      );
+      query.hiddenGlobalTracks =
+        encodeUintSetForUrlComponent(hiddenGlobalTracks) || undefined;
+    }
+    if (query.hiddenLocalTracksByPid) {
+      query.hiddenLocalTracksByPid = (query.hiddenLocalTracksByPid: string)
+        .split('~')
+        .map(pidAndTracks => {
+          const [pid, ...tracks] = pidAndTracks.split('-');
+          const hiddenTracks = new Set(tracks.map(s => +s));
+          return `${pid}-${encodeUintSetForUrlComponent(hiddenTracks)}`;
+        })
+        .join('~');
+    }
+    if (query.localTrackOrderByPid) {
+      query.localTrackOrderByPid = (query.localTrackOrderByPid: string)
+        .split('~')
+        .map(pidAndTracks => {
+          const [pid, ...tracks] = pidAndTracks.split('-');
+          const trackOrder = tracks.map(s => +s);
+          return `${pid}-${encodeUintArrayForUrlComponent(trackOrder)}`;
+        })
+        .join('~');
+    }
+
+    // In this version, uintarray-encoding started supporting a range syntax:
+    // Instead of "abcd" we now support "awd" as a shortcut.
+    // This affects all transform-related uintarrays in the URL. However, we don't
```

Comment:
nit: can you just add the word "also affects"? Initially I thought this comment was about the previous implementation...

Reply:
I rephrased this a bit to be clearer.